libbeat/idxmgmt/std.go

```diff
@@ -254,6 +298,22 @@ func (m *indexManager) Setup(loadTemplate, loadILM LoadMode) error {
 	return nil
 }
 
+func (m *indexManager) setupWithILM() (bool, error) {
+	var err error
+	withILM := m.support.st.withILM.Load()
+	if !withILM {
+		withILM, err = m.ilm.Enabled()
+		if err != nil {
+			return false, err
+		}
+		if withILM {
+			// mark ILM as enabled in indexState
+			m.support.st.withILM.CAS(false, true)
```

Comment:
Where is the timestamp set?

Reply:
Not sure what you mean. I moved it from https://github.com/elastic/beats/pull/12132/files#diff-a7b20900159263d87105489854d4d628L202, related to https://github.com/elastic/beats/pull/12132/files#diff-a7b20900159263d87105489854d4d628L184

Reply:
Nevermind, I confused it with some other code.